proposals/007-azure-kms.md

```diff
@@ -0,0 +1,148 @@
+# 007 - Azure KMS Implementation
+
+One of the flagship Kroxylicious Filters is Record Encryption, enabling users to implement encryption-at-rest for their Apache Kafka cluster. To do this securely, we rely on third-party Key Management Systems (KMS) to protect a set of Key Encryption Keys which are used to encrypt the Data Encryption Keys we write into the records. We currently have AWS, Hashicorp Vault and Fortanix DSM implementations.
+
+We should implement an Azure Key Vault integration, so that users can use Azure as their KMS for record encryption.
+
+<!-- TOC -->
+* [007 - Azure KMS Implementation](#007---azure-kms-implementation)
+  * [Background](#background)
+    * [Key Vault Types](#key-vault-types)
+    * [Key Types & Wrapping Algorithms](#key-types--wrapping-algorithms)
+    * [Authentication With Key Vault](#authentication-with-key-vault)
+    * [National Clouds](#national-clouds)
+    * [TLS](#tls)
+    * [DEK Generation](#dek-generation)
+    * [Key Identifiers](#key-identifiers)
+  * [Proposed Initial Implementation](#proposed-initial-implementation)
+    * [Configuration](#configuration)
+<!-- TOC -->
+
+## Background
+
+> Azure Key Vault is a cloud service for securely storing and accessing secrets. A secret is anything that you want to tightly control access to, such as API keys, passwords, certificates, or cryptographic keys.
```

Comment:
Like AWS, Microsoft do an on-premise version of Azure deployable to your own datacentre.